### DISCUSSION TITLE
Vercel doesn’t detect middleware.ts in a monorepo

### DISCUSSION THREAD

**Discussion by ivoilic** [vercel/vercel#9563] [Help]
I'm trying to use Vercel Edge Middleware with a Solid Start project. This works totally fine in a repo by itself but when it's in a turborepo monorepo suddenly Vercel doesn't detect middleware.ts in the root of the project. I've tried moving it to the root of the entire monorepo instead and various other locations in the project but no luck. Is this broken, do I need configure something in vercel.json, or is there something I'm doing wrong?

**Comment by zipme** [comment]
We face the same issue, A remix app inside a monorepo. What's strange is, we can see the middleware function being deployed to the preview deployment but not production one. But still, the middleware is not executed there.

**Comment by sync2920** [comment]
I am facing a similar problem.
In my case, I am not using monorepo, but I am using remix run to build my application.
For some reason I can only see that the middleware is deployed only in the preview environment.
But the middleware does not run.
